lib/codegen/src/isa/x86/abi.rs

```diff
@@ -488,9 +488,29 @@ fn insert_common_prologue(
             // Simply decrement the stack pointer.
             pos.ins().adjust_sp_down_imm(Imm64::new(stack_size));
         }
+
+        // Check if there is a special stack limit parameter. If there is
+        // then insert stack check.
+        if let Some(stack_limit) = pos.func.special_param(ArgumentPurpose::StackLimit) {
+            insert_stack_check(pos, stack_limit);
```

Comment:
This is essentially same as `stack_check` instruction in the sense that it also does the check after the stack frame has been already opened.

How should I proceed here? Will Something like this

```
a = sp + imm(stack_size)
f = ifcmp a, stack_limit
trapif gte f, stack_ovf
```

do?

Reply:
It's `sp - imm(stack_size)` and `lte` because the stack grows down, but otherwise, yes, that's the idea. But you may need to reorganize the surrounding code a little, so that if we're also doing probes, we do the limit check before the probes.

Reply:
Ah, yeah, you're entirely right. The thing that the [`stack_check` used the-stack-grows-up](https://github.com/pepyakin/cretonne/blob/ff11982b9c923a8423b1f71bbcf1d4e9806d980c/lib/codegen/src/legalizer/mod.rs#L297) confused me.

On the second thought, we can't use `stack_size` alone (At least if we want to guarantee that we never modify the stack below the given `StackLimit`). Because it doesn't account for the size required for RBP, all CSRs and potentially stack space needed for `probestack`.